cmd/imagec/imagec.go

```diff
@@ -296,7 +304,115 @@ func ImagesToDownload(manifest *Manifest, storeName string) ([]*ImageWithMeta, e
 		}
 	}
 
-	return images, nil
+	return images, imageLayer, nil
+}
+
+// Will updated metadata based on the current image being requested
+//
+// Currently on tags are updated, but once imagec supports pulling by digest
+// then digest will need to be updated as well
+func updateImageMetadata(imageLayer *ImageWithMeta, manifest *Manifest) error {
+	// if standalone no need to continue
+	if options.standalone {
+		return nil
+	}
+	// updated Image slice
+	var updatedImages []*ImageWithMeta
+
+	// assuming calls to the portlayer are cheap, lets get all existing images
+	existingImages, err := ListImages(imageLayer.Store, []*ImageWithMeta{})
+	if err != nil {
+		return fmt.Errorf("updateImageMetadata failed to obtain list of images: %s", err)
+	}
+
+	// iterate overall the images and remove the tag that was just downloaded
+	for id, existingImage := range existingImages {
+
+		imageMeta := &metadata.ImageConfig{}
+		// tag / digest info only resides in the metadata so must unmarshall meta to determine
+		if err := json.Unmarshal([]byte(existingImage.Metadata[MetaDataKey]), imageMeta); err != nil {
+			return fmt.Errorf("updateImageMetadata failed to get existing metadata: Layer(%s) %s", id, err)
+		}
+
+		// if this isn't an image we can skip..
+		if imageMeta.ImageID == "" {
+			continue
+		}
+
+		// is this the same repo (busybox, etc)
+		if imageMeta.Name == options.image {
+			var action int
+			// determine action
+			if id == imageLayer.ID {
+				// add to the array if this layer is the requested layer
+				action = Add
+			} else {
+				// remove from array if this layer isn't the requested layer
+				action = Remove
+			}
+			// TODO: imagec doesn't support pulling by digest - add digest support once
+			// issue 1186 is implemented
+			if newTags, ok := arrayUpdate(manifest.Tag, imageMeta.Tags, action); ok {
+				imageMeta.Tags = newTags
+
+				// We need the parent id this is the layer ID of the parent disk
+				// Parent is returned from the PL as a full URL, so we need
+				// to slice the string (also a slice) get the ID -- since the parent string is a pointer
+				// we need to dereference
+				layerParent := (*existingImage.Parent)[strings.LastIndex(*existingImage.Parent, "/"):len(*existingImage.Parent)]
```

Comment:
why are we doing this instead of using existingImage.Parent directly? I presume there's a reason but I cannot determine what it is, from inspection or the comment.